### ISSUE TITLE
Segfault on sema checks 

### ISSUE THREAD

**Issue by LuisAyuso** [insieme/insieme#19]
Type_checks line 846, this returns a null ptr if the field does not exist.
further use of the var is compromised

**Comment by HerbertJordan** [comment]
Right, sorry. 

Please create a test case making it fail and then add something like
```C++
if (!requiredType) {
   add(res, Message(address,
           EC_TYPE_INVALID_INITIALIZATION_EXPR,
           format(
                   "No member %s in struct type %s", 
                   cur->getName().c_str(), 
                   toString(*structType).c_str()
           ),
           Message::ERROR)
    );
}
```
in line 848. This should fix it. I have no resources at the moment to build insieme.

**Comment by stefanmoosbrugger** [comment]
Fixed in commit a3fee3ff64598befe67264f875705593939b428b
